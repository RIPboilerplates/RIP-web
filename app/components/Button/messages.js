import { defineMessages } from 'react-intl'

/*
 * Button Messages
 *
 * This contains all the text for the Button component.
 */
export default defineMessages({
  button: {
    id:             'app.components.Button.button',
    defaultMessage: 'Log Out Action!',
  },
})
