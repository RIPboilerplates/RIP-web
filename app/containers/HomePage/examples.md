[//]: # 'For more details see https://react-styleguidist.js.org/docs/documenting.html#usage-examples-and-readme-files'

Default example:

```js
<HomePage />
```
