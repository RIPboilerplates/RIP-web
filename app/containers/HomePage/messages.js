import { defineMessages } from 'react-intl'

/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
export default defineMessages({
  pageTitle: {
    id:             'app.containers.HomePage.pageTitle',
    defaultMessage: 'LoginPage',
  },
  pageDescription: {
    id:             'app.containers.HomePage.pageDescription',
    defaultMessage: 'Where some sweet stuff happens',
  },
  header: {
    id:             'app.containers.HomePage.header',
    defaultMessage: 'This is the HomePage container!',
  },
})
