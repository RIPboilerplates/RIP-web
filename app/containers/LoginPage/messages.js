import { defineMessages } from 'react-intl'

/*
 * LoginPage Messages
 *
 * This contains all the text for the LoginPage component.
 */
export default defineMessages({
  pageTitle: {
    id:             'app.containers.LoginPage.pageTitle',
    defaultMessage: 'LoginPage',
  },
  pageDescription: {
    id:             'app.containers.LoginPage.pageDescription',
    defaultMessage: 'Page for logging in',
  },
  button: {
    id:             'app.containers.LoginPage.button',
    defaultMessage: 'LOGIN',
  },
})
