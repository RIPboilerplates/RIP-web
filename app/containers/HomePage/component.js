import React from 'react'
import { FormattedMessage } from 'react-intl'
import messages from './messages'
import Styled from './styles'

/**
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 */
export default class HomePage extends React.PureComponent {
  render() {
    return (
      <Styled.Container>
        <h1>
          <FormattedMessage {...messages.header} />
        </h1>
      </Styled.Container>
    )
  }
}
