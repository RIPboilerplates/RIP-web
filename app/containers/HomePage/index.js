/**
 * Exports the component for HomePage
 */
import HomePage from './component'
export default HomePage
