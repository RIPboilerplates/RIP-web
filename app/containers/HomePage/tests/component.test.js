import React from 'react'
import { FormattedMessage } from 'react-intl'
import { shallow } from 'enzyme'

import HomePage from '../component'
import messages from '../messages'

describe('HomePage Component', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(
      <HomePage />
    )
    expect(renderedComponent.contains(
      <FormattedMessage {...messages.header} />
    )).toEqual(true)
  })
})
