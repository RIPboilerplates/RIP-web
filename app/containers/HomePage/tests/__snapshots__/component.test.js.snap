// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component Snapshot Renders default 1`] = `
<div
  className="styles__Container-eZxbLb gPOwxP"
>
  <h1>
    <span>
      This is the HomePage container!
    </span>
  </h1>
  <button
    className="styles__Container-eZxbLb fgfewV"
    onClick={[MockFunction]}
  >
    <span>
      Log Out Action!
    </span>
  </button>
</div>
`;
